tournamentcontrol/competition/tests/test_cross_division_matches.py

```diff
@@ -0,0 +1,216 @@
+        
+        # Add both teams to the same pool initially
+        pool1.teams.add(self.team1_div1, self.team2_div1)
+        
+        match = factories.MatchFactory(
+            stage=self.stage1,
+            stage_group=pool1,
+            ignore_group_validation=False,
+            home_team=self.team1_div1,
+            away_team=self.team2_div1
+        )
+        
+        # Should accept teams from the same pool when flag is False
+        form_data = {
+            'stage_group': pool1.pk,
+            'home_team': self.team1_div1.pk,
+            'away_team': self.team2_div1.pk,  # Same pool
+            'ignore_group_validation': '0',
+            'include_in_ladder': '1',
+        }
+        
+        form = MatchEditForm(instance=match, data=form_data)
+        self.assertTrue(form.is_valid(), form.errors)
+        
+        # Verify the flag is correctly set to False
+        saved_match = form.save()
+        self.assertFalse(saved_match.ignore_group_validation)
+
+    def test_undecided_team_validation_respects_flag(self):
+        """Test that undecided team validation also respects the ignore_group_validation flag."""
+        # Create undecided teams for different stages
+        undecided1 = factories.UndecidedTeamFactory(stage=self.stage1)
+        undecided2 = factories.UndecidedTeamFactory(stage=self.stage2)
+        
+        match = factories.MatchFactory(
+            stage=self.stage1,
+            ignore_group_validation=True
+        )
+        
+        # Should allow undecided teams from different stages when flag is set
+        form_data = {
+            'home_team_undecided': undecided1.pk,
+            'away_team_undecided': undecided2.pk,  # Different stage
+            'ignore_group_validation': '1',
+            'include_in_ladder': '1',
+        }
+        
+        form = MatchEditForm(instance=match, data=form_data)
+        self.assertTrue(form.is_valid(), form.errors)
+
+    def test_match_string_representation_includes_validation_status(self):
+        """Test that matches clearly indicate their non-standard status."""
+        # Create a standard match
+        standard_match = factories.MatchFactory(
+            stage=self.stage1,
+            home_team=self.team1_div1,
+            away_team=self.team2_div1,
+            ignore_group_validation=False
+        )
+        
+        # Create a cross-division match
+        cross_div_match = factories.MatchFactory(
+            stage=self.stage1,
+            home_team=self.team1_div1,
+            away_team=self.team1_div2,
+            ignore_group_validation=True
+        )
+        
+        # Both should have valid string representations
+        self.assertIn("Team 1 Div A", str(standard_match))
+        self.assertIn("Team 2 Div A", str(standard_match))
+        
+        self.assertIn("Team 1 Div A", str(cross_div_match))
+        self.assertIn("Team 1 Div B", str(cross_div_match))
```

Comment:
Never use string representations.

You can `assertEqual` and validate actual field values.